code/game/objects/effects/mines.dm

```diff
@@ -5,20 +5,26 @@
 	anchored = TRUE
 	icon = 'icons/obj/items_and_weapons.dmi'
 	icon_state = "uglymine"
-	var/triggered = 0
+	var/triggered = FALSE
+	/// whether we're deleting ourself or if we're just nullspacing ourself and letting something else delete us (shrapnel mines)
+	var/del_self = TRUE

```

Comment:
AnturK seemed to have a different idea than me on where this should be handled.
Poke me on discord if you want help on it, but basically the component needs to register on the parent's `COMSIG_PARENT_PREQDELETED`, stopping the deletion and handling it there, moving it to nullspace and deleting when over.

Reply:
What the- I swear I reworked it to be good and pushed it but it's not here..... I must've messed up the got push, I'll check it soon